parsl/dataflow/dflow.py

```diff
@@ -161,13 +161,13 @@ def __init__(self, config: Config) -> None:
                                  workflow_info))
 
         if config.checkpoint_files is not None:
-            checkpoints = self.load_checkpoints(config.checkpoint_files)
+            checkpoint_files = config.checkpoint_files
         elif config.checkpoint_files is None and config.checkpoint_mode is not None:
-            checkpoints = self.load_checkpoints(get_all_checkpoints(self.run_dir))
+            checkpoint_files = get_all_checkpoints(self.run_dir)
         else:
```

Comment:
Not the focus of this PR per se, but I observe that given the first `if` condition, the first `elif` clause is redundant.  Of _course_ `checkpoint_files is None`.